dotCMS/src/main/java/com/dotcms/filters/interceptor/saml/SamlWebInterceptor.java

```diff
@@ -0,0 +1,557 @@
+
+                    if (isLogoutNeed &&
+                            this.samlWebUtils.isLogoutRequest(request.getRequestURI(), this.getLogoutPathArray(identityProviderConfiguration))) {
+
+                        if (this.doLogout(response, request, session, identityProviderConfiguration)) {
+
+                            Logger.info(this, "SAML Logout DONE!!!");
+                        }
+                    }
+                } else {
+
+                    Logger.info(this, "No idpConfig for site '" + request.getServerName()
+                            + "'. No SAML filtering for this request: " + request.getRequestURI());
+                }
+            }
+        } catch (final Exception exception) { // todo: better error handling?
+
+            Logger.error(this,  "Error [" + exception.getMessage() + "] Unable to get idpConfig for Site '" +
+                    request.getServerName() + "'. Incoming URL: " + request.getRequestURL(), exception);
+            response.sendError(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
+            return Result.SKIP_NO_CHAIN;
+        } finally {
+            if (null != identityProviderConfiguration) {
+                identityProviderConfiguration.destroy();
+            }
+        }
+
+        return Result.NEXT;
+    } // intercept.
+
+    private void doAuthentication(final HttpServletRequest request,
+                                  final HttpServletResponse response,
+                                  final HttpSession session,
+                                  final IdentityProviderConfiguration identityProviderConfiguration) throws IOException {
+
+        Logger.debug(this, ()-> "There's no logged-in user. Processing SAML request...");
+        this.doRequestLoginSecurityLog(request, identityProviderConfiguration);
+
+        final String originalRequest = request.getRequestURI() +
+                (UtilMethods.isSet(request.getQueryString())?
+                    "?" + request.getQueryString() : StringUtils.EMPTY);
+
+        final String redirectAfterLogin = UtilMethods.isSet(request.getParameter(REFERRER_PARAMETER_KEY))
+                ?request.getParameter(REFERRER_PARAMETER_KEY) :
+                // this is safe, just to make a redirection when the user get's logged.
+                originalRequest;
+
+        Logger.debug(this.getClass(),
+                ()-> "Executing SAML Login Redirection with request: " + redirectAfterLogin);
+
+        // if we don't have a redirect yet
+        if (null != session) {
+
+            session.setAttribute(WebKeys.REDIRECT_AFTER_LOGIN, redirectAfterLogin);
+            session.setAttribute(ORIGINAL_REQUEST,             originalRequest);
+        }
+
+        try {
+            // this will redirect the user to the IdP Login Page.
+            DotSamlProxyFactory.getInstance().samlAuthenticationService()
+                    .authentication(request, response, identityProviderConfiguration);
+        } catch (Exception exception) {
```

Comment:
Codacy found an issue: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5707914)